SQLite/Core/ConnectionPool.swift

```diff
@@ -0,0 +1,177 @@
+//
+// SQLite.swift
+// https://github.com/stephencelis/SQLite.swift
+// Copyright © 2014-2015 Stephen Celis.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+import Dispatch
+import CSQLite
+
+
+private let vfsName = "unix-excl"
+
+
+/// Connection pool delegate
+public protocol ConnectionPoolDelegate {
```

Comment:
Ah, right. That makes sense! I wonder if the same thing could be accomplished internally/privately for most of the connection's helper functions (along with an additional public callback interface). _I.e._ when `createFunction`, etc., are called, that list of setup closures is appended to, called against all existing connections, and the entire list is called against newly-spawned connections.

Something like this:

``` swift
let pool = try ConnectionPool()
pool.foreignKeys = true
// pool.onConnect { db in db.run("PRAGMA foreign_keys = 1") }
```

I can see this being slightly less flexible with the helper function abstraction, but the simple callback block in isolation could be good enough if we don't define those helpers directly on `ConnectionPool`.

My main hesitation with adding a delegate API prematurely is that I just haven't seen the delegate pattern used as much in pure Swift libraries. I wonder if it's the appropriate API and if we should consider alternatives before committing to the public interface.